### DISCUSSION TITLE
`npm outdated` colors

### DISCUSSION THREAD

**Discussion by venables** [npm/feedback#620] [CLI] [answered]
When running `npm outdated`, packages that can easily be upgraded (patch-level) show up in the color red, suggesting you should stop.  Minor-level upgrades are yellow, which is expected.  A green colored patch-level upgrade would make more sense.

It seems logical that a red = major, yellow = minor. green = patch concept would apply to npm users attempting to upgrade.

**Comment by ljharb** [comment]
Red and green aren’t differentiable to those with colorblindness, so the two together might not be the best choice,

**Reply by venables** [reply]
Well thats an easy fix. Use green for patch level and dont use red.

**Reply by ljharb** [reply]
Why is patch safer/better than minor? There’s really only two options: dangerous (major) and probably not dangerous (minor or patch).

**Reply by venables** [reply]
It appears that you're trolling this issue, @ljharb.  But my suggestion is really only about patch level colors and their current RED color.  Patch-level fixes in semver should be able to be applied to any codebase without any issue.  Thus my suggestion for green, rather than red.   Minor upgrades require more thought & should continue to be yellow as they are.  And majors require more attention, as expected, so should be red.

**Reply by ljharb** [reply]
No, patch and minor are identically as safe - either has the same likelihood of bugs, and both have the same intention of not breaking anything. So yes, patch and minor should have the same color.

**Reply by venables** [reply]
@ljharb then you agree! It shouldn't be yellow for minor and red for patch.

For you, it should be the same color.

My suggestion is just that we do yellow for minor (existing color) and green for patch, that's all.

**Comment by MylesBorins** [comment, marked as answer]
@venables if you would like to suggest a specific change to the CLI the proper next step is to open an RFC or discussion at https://github.com/npm/rfcs
